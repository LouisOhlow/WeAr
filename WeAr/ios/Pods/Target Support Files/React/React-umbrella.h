#ifdef __OBJC__
#import <UIKit/UIKit.h>
#else
#ifndef FOUNDATION_EXPORT
#if defined(__cplusplus)
#define FOUNDATION_EXPORT extern "C"
#else
#define FOUNDATION_EXPORT extern
#endif
#endif
#endif

#import "React/Base/RCTAssert.h"
#import "React/Base/RCTBridge.h"
#import "React/Base/RCTBridgeDelegate.h"
#import "React/Base/RCTBridgeModule.h"
#import "React/Base/RCTCache.h"
#import "React/Base/RCTConvert.h"
#import "React/Base/RCTDefines.h"
#import "React/Base/RCTEventDispatcher.h"
#import "React/Base/RCTFPSGraph.h"
#import "React/Base/RCTFrameUpdate.h"
#import "React/Base/RCTInvalidating.h"
#import "React/Base/RCTJavaScriptExecutor.h"
#import "React/Base/RCTJavaScriptLoader.h"
#import "React/Base/RCTJSMethodRegistrar.h"
#import "React/Base/RCTKeyboardObserver.h"
#import "React/Base/RCTKeyCommands.h"
#import "React/Base/RCTLog.h"
#import "React/Base/RCTModuleData.h"
#import "React/Base/RCTModuleMap.h"
#import "React/Base/RCTModuleMethod.h"
#import "React/Base/RCTPerformanceLogger.h"
#import "React/Base/RCTPerfStats.h"
#import "React/Base/RCTProfile.h"
#import "React/Base/RCTRootView.h"
#import "React/Base/RCTSparseArray.h"
#import "React/Base/RCTTouchHandler.h"
#import "React/Base/RCTURLRequestDelegate.h"
#import "React/Base/RCTURLRequestHandler.h"
#import "React/Base/RCTUtils.h"
#import "React/Executors/RCTContextExecutor.h"
#import "React/Executors/RCTWebViewExecutor.h"
#import "React/Layout/Layout.h"
#import "React/Modules/RCTAccessibilityManager.h"
#import "React/Modules/RCTAlertManager.h"
#import "React/Modules/RCTAppState.h"
#import "React/Modules/RCTAsyncLocalStorage.h"
#import "React/Modules/RCTDevLoadingView.h"
#import "React/Modules/RCTDevMenu.h"
#import "React/Modules/RCTExceptionsManager.h"
#import "React/Modules/RCTPointAnnotation.h"
#import "React/Modules/RCTRedBox.h"
#import "React/Modules/RCTSourceCode.h"
#import "React/Modules/RCTStatusBarManager.h"
#import "React/Modules/RCTTiming.h"
#import "React/Modules/RCTUIManager.h"
#import "React/Views/RCTActivityIndicatorViewManager.h"
#import "React/Views/RCTAnimationType.h"
#import "React/Views/RCTAutoInsetsProtocol.h"
#import "React/Views/RCTBorderDrawing.h"
#import "React/Views/RCTComponent.h"
#import "React/Views/RCTComponentData.h"
#import "React/Views/RCTConvert+CoreLocation.h"
#import "React/Views/RCTConvert+MapKit.h"
#import "React/Views/RCTDatePickerManager.h"
#import "React/Views/RCTMap.h"
#import "React/Views/RCTMapManager.h"
#import "React/Views/RCTModalHostView.h"
#import "React/Views/RCTModalHostViewController.h"
#import "React/Views/RCTModalHostViewManager.h"
#import "React/Views/RCTNavigator.h"
#import "React/Views/RCTNavigatorManager.h"
#import "React/Views/RCTNavItem.h"
#import "React/Views/RCTNavItemManager.h"
#import "React/Views/RCTPicker.h"
#import "React/Views/RCTPickerManager.h"
#import "React/Views/RCTPointerEvents.h"
#import "React/Views/RCTProgressViewManager.h"
#import "React/Views/RCTScrollableProtocol.h"
#import "React/Views/RCTScrollView.h"
#import "React/Views/RCTScrollViewManager.h"
#import "React/Views/RCTSegmentedControl.h"
#import "React/Views/RCTSegmentedControlManager.h"
#import "React/Views/RCTShadowView.h"
#import "React/Views/RCTSlider.h"
#import "React/Views/RCTSliderManager.h"
#import "React/Views/RCTSwitch.h"
#import "React/Views/RCTSwitchManager.h"
#import "React/Views/RCTTabBar.h"
#import "React/Views/RCTTabBarItem.h"
#import "React/Views/RCTTabBarItemManager.h"
#import "React/Views/RCTTabBarManager.h"
#import "React/Views/RCTTextDecorationLineType.h"
#import "React/Views/RCTView.h"
#import "React/Views/RCTViewControllerProtocol.h"
#import "React/Views/RCTViewManager.h"
#import "React/Views/RCTWebView.h"
#import "React/Views/RCTWebViewManager.h"
#import "React/Views/RCTWrapperViewController.h"
#import "React/Views/UIView+Private.h"
#import "React/Views/UIView+React.h"

FOUNDATION_EXPORT double ReactVersionNumber;
FOUNDATION_EXPORT const unsigned char ReactVersionString[];

