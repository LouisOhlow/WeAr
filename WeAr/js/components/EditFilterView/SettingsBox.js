import React from 'react';
import {
  View, Image, TouchableOpacity, StyleSheet,
} from 'react-native';
import COLORS from '../../res/colors';
import Headline1 from '../basics/Headline1';

export default function SettingsBox({ navigate, title, image }) {
  return (
    <View style={styles.setting}>
      <TouchableOpacity onPress={() => { navigate(); }} style={styles.box}>
        <Image
          style={styles.overlay}
          source={image}
        />
        <View style={styles.overlay} />
        <Headline1 text={title} />
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  setting: {
    width: '80%',
    height: 180,
    margin: 27,
  },
  box: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    justifyContent: 'center',
  },
  overlay: {
    width: '100%',
    height: 180,
    position: 'absolute',
    backgroundColor: COLORS.semiblack,
    borderRadius: 50,
  },
});
