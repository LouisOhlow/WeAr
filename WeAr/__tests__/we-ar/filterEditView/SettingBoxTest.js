import React from 'react';
import renderer from 'react-test-renderer';
import SettingsBox from '../../../js/components/EditFilterView/SettingsBox';

describe('Snapshot tests of the settings header', () => {

    it('only with a back button', () => {
        const tree = renderer
          .create( 
          <SettingsBox 
            navigate={() => {}} 
            title="CHANGE TEST COLOR" 
            image={require('../../../js/drawables/colored_avocado.png')} />)
          .toJSON();
        expect(tree).toMatchSnapshot();
    });
  });
