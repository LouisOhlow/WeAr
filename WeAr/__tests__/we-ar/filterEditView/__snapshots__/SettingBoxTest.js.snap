// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests of the settings header only with a back button 1`] = `
<View
  style={
    Object {
      "height": 180,
      "margin": 27,
      "width": "80%",
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "bottom": 0,
        "justifyContent": "center",
        "left": 0,
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../js/drawables/colored_avocado.png",
        }
      }
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "borderRadius": 50,
          "height": 180,
          "position": "absolute",
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "borderRadius": 50,
          "height": 180,
          "position": "absolute",
          "width": "100%",
        }
      }
    />
    <View>
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 20,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        CHANGE TEST COLOR
      </Text>
    </View>
  </View>
</View>
`;
